pygpudrive/env/base_env.py

```diff
@@ -110,6 +110,29 @@ def _setup_environment_parameters(self):
 
         params = self._set_collision_behavior(params)
         params = self._set_road_reduction_params(params)
+        
+        # Map entity types to integers  
+        self.ENTITY_TYPE_TO_INT = {
+            gpudrive.EntityType._None: 0,
+            gpudrive.EntityType.RoadEdge: 1,
+            gpudrive.EntityType.RoadLine: 2,
+            gpudrive.EntityType.RoadLane: 3,
+            gpudrive.EntityType.CrossWalk: 4,
+            gpudrive.EntityType.SpeedBump: 5,
+            gpudrive.EntityType.StopSign: 6,
+            gpudrive.EntityType.Vehicle: 7,
+            gpudrive.EntityType.Pedestrian: 8,
+            gpudrive.EntityType.Cyclist: 9,
+            gpudrive.EntityType.Padding: 10,
+        }
+        self.MIN_OBJ_ENTITY_ENUM = min(list(self.ENTITY_TYPE_TO_INT.values()))
+        self.MAX_OBJ_ENTITY_ENUM = max(list(self.ENTITY_TYPE_TO_INT.values()))
+        self.ROAD_MAP_OBJECT_TYPES = 7 # (enums 0-6)
+        self.ROAD_OBJECT_TYPES = 4 # (enums 7-10)
```

Comment:
I think there is some redundancy here:
- An enum is an alias for an integer, so there isn't a need to map the names to their underlying representation.
- If you split `ENTITY_TYPE_TO_INT` into two lists- one for agent entities and one for road map entities- you can eliminate `ROAD_MAP_OBJECT_TYPES ` and `ROAD_OBJECT_TYPES` because they would equal the length of the corresponding lists.
```suggestion
	self.ROAD_MAP_ENTITY_TYPES = [
            gpudrive.EntityType._None,
            gpudrive.EntityType.RoadEdge,
            gpudrive.EntityType.RoadLine,
            gpudrive.EntityType.RoadLane,
            gpudrive.EntityType.CrossWalk,
            gpudrive.EntityType.SpeedBump,
            gpudrive.EntityType.StopSign,
	self.AGENT_ENTITY_TYPES	= [
            gpudrive.EntityType.Vehicle,
            gpudrive.EntityType.Pedestrian,
            gpudrive.EntityType.Cyclist,
            gpudrive.EntityType.Padding,
	]
        self.MIN_OBJ_ENTITY_ENUM = min(self.ROAD_MAP_ENTITY_TYPES + self.AGENT_ENTITY_TYPES)
        self.MAX_OBJ_ENTITY_ENUM = max(self.ROAD_MAP_ENTITY_TYPES + self.AGENT_ENTITY_TYPES)
```

Moreover, I'm not sure these should be instance variables.

Reply:
Thanks for pointing that out, I made the changes

Reply:
Question: I still need to map the enums to integers. How would you suggest to do this? 

Instead of the dictionary above, I could do something like: 
```Python
{
     gpudrive.EntityType._None: int( gpudrive.EntityType._None),
...
}
```

Reply:
You can use a comprehension instead (ie. `{(k, int(k)) for k in self.ENTITY_TYPES}`) but looking through the functions `one_hot_encode_*`, I don't immediately see where this mapping is necessary.

Reply:
True, and it is used in the torch environment, e.g. see line 269 in `env_torch.py` in this PR